### ISSUE TITLE
Not checking games in external storage

### ISSUE THREAD

**Issue by omakrivic** [silverthornne/steamdeckbash#1]
Ive been using your script and is working perfectly fine but is not recognizing the games from the external storage or i dont know if is going to just recognize one store every time as Ive been using when i had my sd card and the storage connected through the dock station.

**Comment by silverthornne** [comment]
I'm going to need some help with that issue because I personally don't use external storage. You're referring to an M2 drive in the JSAUX dock if I'm not mistaken, right? It can probably be tackled by forking the script and changing the variable for the SD Card slot location to the path for the external storage location, but that would be a separate script. Ideally, the same script should handle both.

Line 43 needs to be modified on a forked script.

For the SD card, it reads:

sCardPath="/run/media/mmcblk0p1"

It needs to be modified for the path that the external storage mounts to.

**Comment by silverthornne** [comment]
If you could provide me with the mount path for your external storage, I can probably do a new script that asks if you want to look into SD or external storage, but it would be specific to the path in your device for that drive. As I haven't used external storage on my Deck, I'm not sure how it behaves with the way Gaming Mode mounts it.

**Comment by omakrivic** [comment]
So I think it should be just as simple as ask with a variable for another path maybe as you said with another script and the user can copy/paste the path of the directory and you can describe like add the path "xxxx/steam/abc/abc" and should work. I mean this because even if I add my path here is a custom as per each external storage you need to format it, name it and then add it to steam manually

**Comment by silverthornne** [comment]
If you need to format the external storage with its own custom path then yes, you'd need to have a script for each one of those. Or, the script could start by asking the path for the drive. That would be your forked script. It does add some risk in that you may enter the wrong path, but the script won't return anything if the path doesn't work when building the menu so it wouldn't be a catastrophic fail.

You'd need to start the script with something like:

echo "----------Please enter the path to the drive to scan for games.----------"
read -p "-> " sCardPath

Of course, you'd need to remove the sCardPath="/run/media/mmcblk0p1" line because it's now being set with that read command.

Also, to avoid headaches, the path to the external drive shouldn't have spaces, and you shouldn't include the last / in the path name. Notice that the way the variable is set for sCardPath it doesn't end in a /.

**Comment by omakrivic** [comment]
Ok let me modify the script and add something ill do some test and will let you know in case you want to add it. By the way Ive been thinking maybe if you want i can open a new issue about a feature to improve the script in which for example my sd is 1TB and now im taking 1 by 1 the games so instead of going 1 by 1 just have an option to pull all the games from the sd and move the folder with a loop

**Comment by silverthornne** [comment]
I've actually been thinking about a looping script that does that for a while, but haven't sat down to write it yet. I'll probably look into that during the Xmas break as I just joined a company that shuts down for about a week for Xmas.

Anyway, I think I have a better solution for the external storage situation: a script that asks the user which storage to use when moving data!

No promises on a timeline, but I should be able to build a menu that looks at your storage partitions and asks which to use to check for games and move the data to.

If you have a chance, could you please run this command quickly and tell me if it displays your external storage as one of the results?

`df -h | awk 'FNR > 3 {print $6}'`

My concept is just to add a menu list that asks which of those results to use and then proceed with the script as usual. The reason I'd do it as a new script instead of modifying the current one to do it is the name. The current script's name includes "SD"; I shouldn't have limited it that way haha.

**Comment by omakrivic** [comment]
Sorry today i had issues with the M2 that sometimes is tricky and ive been fixing it. So yes using the command line that you mentioned i can see the path for my M2:
![image](https://user-images.githubusercontent.com/7346054/205000226-6c02ceaa-2cd2-439c-a7b3-0f7ae557f6cf.png)
/run/media/deck/Library is my M2

**Comment by silverthornne** [comment]
/run/media/deck/Library?

Ok, it seems the Deck has more partitions than I was expecting; I tested the command on one of my servers, not directly on the Deck :)

So it should probably be FNR > 8 at least? I'll mess around on my deck today to refine it better. I should probably just have cases for partitions we wouldn't want to be on the menu like /run, /var, /etc, and so on and filter them out.

**Comment by silverthornne** [comment]
I added a new script that may work with external storage drives. Give it a go if you're brave enough :)

I can't test it myself, but at least it does work if I choose the microSD card as the mount point. It will list your storage partitions and ask you where to look in for the games.

**Comment by omakrivic** [comment]
Sorry Ive in the middle of something but 6 of December ill check and let you know.
Im from spain but living in Texas thats why im putting the dat for the different time zones

**Comment by silverthornne** [comment]
No hurries. ¡Espero que las estés pasando bien en Texas!

**Comment by omakrivic** [comment]
Gracias! Ok ive tried so far and it seems that maybe is needed another think or you need to tell me which folders is expecting over there as right now with the screenshots attach this is the error im getting:
![image](https://user-images.githubusercontent.com/7346054/206572339-d1b46bc6-8aba-4fcd-ab43-d4a679e32eed.png)
![image](https://user-images.githubusercontent.com/7346054/206572372-45468a81-7abd-4a62-8353-dbc380d4b288.png)

**Comment by silverthornne** [comment]
Of course! If it's the MicroSD card and it was installed from Gaming Mode, Steam places the games in steamapps. However, Steam also uses SteamLibrary if using the Steam desktop client. If I'm following this correctly, you installed the games from the Steam client instead of Gaming Mode and created the Steam library in the external drive. As I don't have an external drive, I failed to consider that scenario, so the script must also search in SteamLibrary and not just on steamapps. Ok, gotta work on that!

**Comment by omakrivic** [comment]
Ok let me know when you update the script and will try again!

**Comment by silverthornne** [comment]
Oh, one other question: inside your internal storage, did Steam create the compatibility and shadercache data for those external storage games in the same path as it does for MicroSD card games (steamapps) or did it create a SteamLibrary path structure for them?

**Comment by omakrivic** [comment]
I dont see anything created

**Comment by silverthornne** [comment]
Can you navigate to /home/deck/.local/share/Steam/steamapps/compatdata and verify if there are directories for the games you have on your external storage? You'd need the AppID for games in your external storage and see if there's a directory for any of those AppIDs in there. Same in /home/deck/.local/share/Steam/steamapps/shadercache.

If there is no local SteamLibrary directory, there's a chance that the compatibility data is in the same place as for games in internal storage and the MicroSD card or that it's always been placed in the same drive as the game. Can you check if there is a compatdata and a shadercache directory in /run/media/deck/Library/SteamLibrary?

**Comment by omakrivic** [comment]
So i checked God of war with the App ID 1593500 and is /home/deck/.local/share/Steam/steamapps/compatdata and /home/deck/.local/share/Steam/steamapps/shadercache but is installed in the M2 the external storage.

And there is no compatdata and a shadercache directory in /run/media/deck/Library/SteamLibrary

**Comment by silverthornne** [comment]
Ok, so it still uses steamapps internally. Good. Ok, with that I can proceed. Thanks!

**Comment by omakrivic** [comment]
Ok will wait until you tell me to try/check anything you need
